app/models/piece.rb

```diff
@@ -5,6 +5,19 @@ class Piece < ActiveRecord::Base
   validates :color, inclusion: { in: %w(black white) }
   validates :type, inclusion: { in: %w(Pawn Rook Bishop Knight King Queen) }
 
+  # Returns "valid" and updates the piece's x and
+  # y position to provided coordinates if move is valid,
+  # otherwise do nothing and return nil.
+  def move(x,y)
+    if valid_move?(x,y)
+      self.x_position = x
+      self.y_position = y
+      return "valid"
```

Comment:
Well it doesn't necessarily need to be a bang method, but I would say that a true/false return indicating whether the move actually happened is completely valid to expect from this method.

This is one of the things that bugs me with rails. Technically this should be a bang method. But it should also be a huh. But really, why the hell did they decide we had to decide? What the crap is wrong with a regular method... /rant